temporal-sdk/src/main/java/io/temporal/internal/common/SearchAttributesUtil.java

```diff
@@ -0,0 +1,182 @@
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.common;
+
+import io.temporal.api.common.v1.Payload;
+import io.temporal.api.common.v1.SearchAttributes;
+import io.temporal.common.converter.DataConverterException;
+import io.temporal.common.converter.DefaultDataConverter;
+import io.temporal.common.converter.SearchAttributesPayloadConverter;
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SearchAttributesUtil {
+
+  public static DefaultDataConverter converter = DefaultDataConverter.newDefaultInstance();
+  private static final Logger log = LoggerFactory.getLogger(SearchAttributesUtil.class);
+
+  public enum SearchAttribute {
+    BatcherNamespace,
+    BatcherUser,
+    BinaryChecksums,
+    CloseTime,
+    ExecutionDuration,
+    ExecutionStatus,
+    ExecutionTime,
+    HistoryLength,
+    RunId,
+    StartTime,
+    StateTransitionCount,
+    TaskQueue,
+    TemporalChangeVersion,
+    WorkflowId,
+    WorkflowType
+  }
+
+  public enum SearchAttributeType {
+    Unspecified,
+    String,
+    Keyword,
+    Int,
+    Double,
+    Bool,
+    Datetime
+  }
+
+  public static Map<String, Payload> serializeToObjectMap(Map<String, Object> searchAttributes) {
+    if (searchAttributes == null) {
+      return null;
+    }
+
+    Map<String, Payload> serializedMap = new HashMap<>();
+    SearchAttributesPayloadConverter dataConverter = new SearchAttributesPayloadConverter();
+    for (Map.Entry<String, Object> item : searchAttributes.entrySet()) {
+      try {
+        serializedMap.put(item.getKey(), dataConverter.toData(item.getValue()).get());
+      } catch (DataConverterException e) {
+        throw new DataConverterException("Cannot serialize key " + item.getKey(), e.getCause());
+      }
+    }
+
+    return serializedMap;
+  }
+
+  public static Map<String, Object> deserializeToObjectMap(SearchAttributes serializedMap) {
+    if (serializedMap == null) {
+      return null;
+    }
+
+    Map<String, Object> deserializedMap = new HashMap<>();
+    for (Entry<String, Payload> attribute : serializedMap.getIndexedFieldsMap().entrySet()) {
+      String key = attribute.getKey();
+      if (!parseSearchAttributes(key, attribute.getValue(), deserializedMap)) {
+        String typeString = attribute.getValue().getMetadataMap().get("type").toStringUtf8();
+        Type javaType = stringToJavaType(typeString);
+        if (javaType == null) {
+          log.error("Error parsing Search Attribute: " + key + " of type " + typeString);
+        } else {
+          deserializedMap.put(
+              key, converter.fromPayload(attribute.getValue(), javaType.getClass(), javaType));
```

Comment:
Why is this here if it belongs to our specialized converter? Why is a default converted is used here that can be redefined instead of the specialized converter?